server/queue/queue-rabbitmq/src/test/java/org/apache/james/queue/rabbitmq/RabbitMQMailQueueTest.java

```diff
@@ -231,4 +235,35 @@ private void dequeueMails(int times) {
         IntStream.rangeClosed(1, times)
             .forEach(Throwing.intConsumer(bucketId -> mailQueue.deQueue().done(true)));
     }
+
+    private static class UpdatableTickingClock extends Clock {
+        private Instant currentInstant;
+
+        UpdatableTickingClock(Instant currentInstant) {
+            this.currentInstant = currentInstant;
+        }
+
+        void setInstant(Instant instant) {
+            currentInstant = instant;
+        }
+
+        @Override
+        public ZoneId getZone() {
+            throw new NotImplementedException("No timezone attached to this clock");
+        }
+
+        @Override
+        public Clock withZone(ZoneId zone) {
+            throw new NotImplementedException("No timezone attached to this clock");
+        }
+
+        @Override
+        public synchronized Instant instant() {
```

Comment:
why ?

Reply:
True, this is no more required on the instant method but on the tick method.